### PR TITLE
Run edmEventSize on step3 and step4 root files produced by running igprof on high pileup workflow 23434.21 and save output.

### PR DESCRIPTION
**Pull request by gartung**
This needs to be merged before #1408 abd #1413 to generate the event size text file for PR profiling comparison.

### CODE REVIEW

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
abort

**Comment by gartung** [comment]
test

File: run-ib-igprof [hunk 1]
```diff
@@ -4,11 +4,16 @@ case $CMS_BOT_DIR in /*) ;; *) CMS_BOT_DIR=$(pwd)/${CMS_BOT_DIR} ;; esac
 WORKFLOWS=$1
 PROFILES=$2
 EVENTS=$3
+HPUWF="23434.21"
 if [ "X$EVENTS" = "X" ] ; then EVENTS=1000; fi
 for prof in ${PROFILES} ; do
   mkdir -p $WORKSPACE/igprof/${prof}
   cd $WORKSPACE/igprof/${prof}
   runTheMatrix.py $WORKFLOWS --ibeos --command "-n $EVENTS --profile $prof" 2>&1 | tee runTheMatrix.log
+  if [[ $WORKFLOWS == *"$HPUWF" ]]; then
```
Comment (smuzaffar) [inline run-ib-igprof:13]: @gartung , this will match any workflow which ends with $HPUWF . Is this what we want here?

File: run-ib-igprof [hunk 2]
```diff
@@ -4,11 +4,16 @@ case $CMS_BOT_DIR in /*) ;; *) CMS_BOT_DIR=$(pwd)/${CMS_BOT_DIR} ;; esac
...
+    edmEventSize -v step3*.root > step3_sizes_${HPUWF}.txt
```
Comment (smuzaffar) [inline run-ib-igprof:14]: is not `runTheMatrix.py` create a subdirectory for every workflow and root files are under that sub-directory? So how these commands are going to find these step3*.root files?

if would suggest to do something like
```
for hpfw in ${HPUWFS} ; do
  for s in step3 step4 ; do
    if [ $(ls -d ${hpwf}_*/${s}_*.root | wc -l) -eq 0 ] ; then continue ; fi 
    edmEventSize -v ${hpwf}_*/${s}_*.root> step4_sizes_${HPUWF}.txt
done
```

File: run-ib-igprof (hunk 1 above)
Comment (gartung) [reply run-ib-igprof:13]: Yes. The jenkins job has WORKFLOWS="-l 23434.21"

File: run-ib-igprof (hunk 2 above)
Comment (gartung) [reply run-ib-igprof:14]: The root files are produced when igprof is run and they should be in the same directory.

Comment (gartung) [reply run-ib-igprof:14]: You are correct. I will update this PR.

File: run-ib-igprof [hunk 3]
```diff
@@ -4,11 +4,20 @@ case $CMS_BOT_DIR in /*) ;; *) CMS_BOT_DIR=$(pwd)/${CMS_BOT_DIR} ;; esac
...
 PROFILES=$2
 EVENTS=$3
+HPUWFS="23434.21"
 if [ "X$EVENTS" = "X" ] ; then EVENTS=1000; fi
 for prof in ${PROFILES} ; do
   mkdir -p $WORKSPACE/igprof/${prof}
   cd $WORKSPACE/igprof/${prof}
   runTheMatrix.py $WORKFLOWS --ibeos --command "-n $EVENTS --profile $prof" 2>&1 | tee runTheMatrix.log
+  for hpwf in $HPUWFS; do
+    if [[ $WORKFLOWS == *"${hpwf}" ]]; then
```
Comment (smuzaffar) [inline run-ib-igprof:14]: @gartung , you do not need this check. The check on line 16 is enough to see if a HPWF was run or not

Comment (gartung) [reply run-ib-igprof]: updated

**Comment by smuzaffar** [comment]
@gartung , so this looks good now. This can be merged ... right? It doe snot depend on any of other PRs

File: run-ib-igprof [hunk 4]
```diff
@@ -4,11 +4,18 @@ case $CMS_BOT_DIR in /*) ;; *) CMS_BOT_DIR=$(pwd)/${CMS_BOT_DIR} ;; esac
 WORKFLOWS=$1
 PROFILES=$2
 EVENTS=$3
+PROFILING_WORKFLOWS=$(echo $(grep "PR_TEST_MATRIX_EXTRAS_PROFILING=" $CMS_BOT_DIR/cmssw-pr-test-config | sed 's|.*=||'), | tr ' ' ','| tr ',' '\n' | grep '^[0-9]' | sort | uniq | tr '\n' ',' | sed 's|,*$||')
```
Comment (smuzaffar) [inline run-ib-igprof:7]: this look very complecatied :-) just use
```
PROFILING_WORKFLOWS=$(grep "PR_TEST_MATRIX_EXTRAS_PROFILING=" $CMS_BOT_DIR/cmssw-pr-test-config | sed 's|.*=||;s|,| |')
```

Comment (gartung) [reply run-ib-igprof]: done

**Comment by smuzaffar** [comment]
looks good, lets merge this and test for igprof job

**Comment by smuzaffar** [comment]
jenkins job https://cmssdt.cern.ch/jenkins/job/ib-run-igprof/12844/console should be using it now

**Pull request merged**